src/storage/store.js

```diff
@@ -401,6 +402,20 @@ export default class Store extends EventTarget {
   }
 
   update(newState, mergeOpts) {
+    // Check if in demo room before changing avatar. This feature is meant to prevent any avatarID changes triggered from an avatar.js panel ONLY. "Change Name & Avatar" button in More Menu is disabled at a UI level.
```

Comment:
Looking at this again, I don't think this is avoiding an avatar update. This is called by the `avatar.js` page and there we don't have any context about the current `hub_id`. We just update the store and that event is recived in any other open document that is using the same local storage origin.

It seems that we are processing that here:
https://github.com/mozilla/hubs/blob/b8f775f10da2fdd2b9515adc992050ca7c3cf986/src/scene-entry-manager.js#L180

And there we can correctly get the `hub_id` so I think we can revert the changes here and add that logic there.